Makefile

```diff
@@ -256,15 +256,27 @@ bootstrap-jsons: $(BOOTSTRAP_GHC_VERSIONS:%=bootstrap-json-%)
 ##############################################################################
 
 .PHONY: users-guide
-users-guide:
+users-guide: ## Build the users guide.
 	$(MAKE) -C doc users-guide
 
 .PHONY: users-guide-requirements
-users-guide-requirements:
+users-guide-requirements: ## Install the requirements for building the users guide.
 	$(MAKE) -C doc users-guide-requirements
 
 ifeq ($(shell uname), Darwin)
 PROCS := $(shell sysctl -n hw.logicalcpu)
 else
 PROCS := $(shell nproc)
 endif
+
+PHONY: help
+help: ## Show the commented targets.
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | \
+	sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+PHONY: help-banner
+help-banner: ## Show the help banner.
+	@echo "===================================================================="
+	@echo "§ all                  make with no arguments also shows this banner"
+	@echo "§ help                 make help will list targets with descriptions"
+	@echo "===================================================================="
```

Comment:
Mind the ⊖.

Reply:
Mind the gap. I've lived in London and ridden on the tube ;-)

Commanding `make` with no target is the same as `make all`. I've made `help-banner` a prerequisite so that the makefile help banner will be printed to draw attention to it.

```
$ make
====================================================================
§ all                  make with no arguments also shows this banner
§ help                 make help will list targets with descriptions
====================================================================
cabal build cabal-install:exes
Warning: this is a debug build of cabal-install with assertions enabled.
Configuration is affected by the following files:
- cabal.project
Build profile: -w ghc-9.10.1 -O1
...
```

Reply:
Actually that's not true: it runs the first target. By convention that target is `all`.  nhttps://www.gnu.org/software/make/manual/make.html#How-Make-Works

Reply:
Good thing `all` is first.

Reply:
That's Phil's usual missing newline at the end of the file.

Reply:
Not something `fix-whitespace` helps with :-(

https://github.com/haskell/cabal/blob/8144494dbf96ae75943896ea25f229924be66728/fix-whitespace.yaml#L104-L105

Reply:
Fixed now.